src/jsMain/kotlin/baaahs/JsMain.kt

```diff
@@ -17,7 +20,17 @@ fun main(args: Array<String>) {
     val contentDiv = document.getElementById("content")
 
     when (mode) {
-        "Simulator" -> SheepSimulator().start()
+        "Simulator" -> {
+            // Instead of starting the simulator directly, pass the JS
+            // a function that it can use to get and start the simulator.
+            // We do this so the JS can create the HTML templates before
+            // the JsDisplay tries to find them.
```

Comment:
Huh, interesting. This won't cause `SheepSimulator` to be re-created every time the react views are re-rendered? I definitely haven't wrapped my head around how React works yet.

Reply:
On the react side, the `getSheepSimulator` function is called in a `useEffect` hook that will only happen once per page load